src/theme/popover/ods-chart-popover.ts

```diff
@@ -344,8 +340,40 @@ export class ODSChartsPopover {
       if (!this.popoverDefinition.getOrCreatePopupInstance) {
         mergeObjects(popoverOptions, {
           tooltip: {
-            position: function (pt: [number, number]) {
-              return [pt[0], pt[1]];
+            position: (pos: any, params: any, dom: any, rect: any, size: any) => {
+              let obj: any = {
+                left: pos[0] - size.contentSize[0] / 2,
+              };
+
+              // console.log('position', pos, obj, size, rect, dom, params);
+              // TODO: define if the tooltip placement is an option or defined by the user
+
+              if (this.popoverConfig.tooltipConfine) {
+                const x = pos[0];
+                const arrowSize = 10;
+                const bottom = pos[1] > size.contentSize[1]; // pos[1] > size.viewSize[1] / 2 or pos[1] > size.contentSize[1] or pos[1] > size.viewSize[1] - size.contentSize[1]
```

Comment:
Future reader, please note that there's a possibility to change the rendering by doing:

```
// pos[1] > size.viewSize[1] / 2 or pos[1] > size.contentSize[1] or pos[1] > size.viewSize[1] - size.contentSize[1]
```